### PR TITLE
elasticsearch - fix issue with bulk indexer closing

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

File: flow/connectors/connelasticsearch/elasticsearch.go [hunk 1]
```diff
@@ -121,25 +121,23 @@ func (esc *ElasticsearchConnector) SyncRecords(ctx context.Context,
...
 		closeHasErrors := false
-		if bulkIndexersHaveShutdown {
+		if !bulkIndexersHaveShutdown {
 			for _, esBulkIndexer := range maps.Values(esBulkIndexerCache) {
 				err := esBulkIndexer.Close(context.Background())
 				if err != nil {
 					esc.logger.Error("[es] failed to close bulk indexer", slog.Any("error", err))
 					closeHasErrors = true
 				}
+				numRecords += int64(esBulkIndexer.Stats().NumFlushed)
```
Comment (serprex) [inline flow/connectors/connelasticsearch/elasticsearch.go:140]:
```suggestion
				numRecords = int64(esBulkIndexer.Stats().NumFlushed)
```

**Review by serprex — APPROVED** [review]

File: flow/connectors/connelasticsearch/elasticsearch.go (hunk 1 above)
Comment (heavycrystal) [reply flow/connectors/connelasticsearch/elasticsearch.go:140]: multiple bulk indexers can be present, one per destination index

File: flow/connectors/connelasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -295,11 +290,12 @@ func (esc *ElasticsearchConnector) SyncRecords(ctx context.Context,
 	if err := esc.FinishBatch(ctx, req.FlowJobName, req.SyncBatchID, lastCheckpoint); err != nil {
 		return nil, err
 	}
+	esc.logger.Info("[es] SyncRecords done", slog.Int64("numRecords", numRecords))
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/connelasticsearch/elasticsearch.go:292]: can keep it around imo

[the remark this answers is not in the document]
Comment (serprex) [reply flow/connectors/connelasticsearch/elasticsearch.go:292]: this should already be logged by activity

**Pull request merged**